frontend/src/user_context.tsx

```diff
@@ -16,15 +16,23 @@ export interface User {
 
 export type Result = 'logged_in' | 'logged_out' | 'error'
 
+/**
+ * The user state is either a user object or `undefined` (waiting for login) or
+ * `null` (anonymous/logged out). This is convienent since we can identify the
+ * state of the user login check if needed, but we can also easily find if a
+ * user object is available by boolean checking.
+ */
+export type UserState = User | undefined | null
```

Comment:
Nu retas du ju bara :P

Reply:
haha - lite, men jag har redan byggt in en massa checkar för `if (user)`, så det var slöare att använda `null` och `undefined` än tydliga och lättlästa värden :D